code/modules/mob/living/silicon/robot/drone/maint_drone.dm

```diff
@@ -151,59 +170,24 @@
 
 	else if(istype(I, /obj/item/card/id) || istype(I, /obj/item/pda))
 		if(stat == DEAD)
-		// Currently not functional, so commenting out until it's fixed to avoid confusion
-			/*if(!config.allow_drone_spawn || emagged || health < -35) //It's dead, Dave.
-				to_chat(user, "<span class='warning'>The interface is fried, and a distressing burned smell wafts from the robot's interior. You're not rebooting this one.</span>")
-				return
+			return
 
-			if(!allowed(I))
-				to_chat(user, "<span class='warning'>Access denied.</span>")
-				return
+		var/confirm = tgui_alert(user, "Using your ID on a Maintenance Drone will shut it down, are you sure you want to do this?", "Disable Drone", list("Yes", "No"))
+		if(confirm == ("Yes") && (user in range(3, src)))
```

Comment:
They should be adjacent, because of the visible message saying they swipe through it, I know they can be annoying and escaping, but it should stay this way
```suggestion
		if(confirm == "Yes" && in_range(user , src))
```

Reply:
Yeah I just removed the commented out code, unless I accidentally did, this should be the same as before